src/_tests/integration/integration-utils.js

```diff
@@ -1,5 +1,38 @@
+const waitForFrame = async url => {
+  const frameLoaded = new Promise(resolve => {
+    const navHandler = async frame => {
+      await frame.url().includes(url) && resolve({ frame, navHandler })
+    }
+    page.on('framenavigated', navHandler)
+  })
+
+  const { frame, navHandler } = await frameLoaded
+  page.removeListener('framenavigated', navHandler)
+  return frame
+}
+
+const findIframe = async page => {
+  const iframeNode = await page.waitForXPath('//*[@role="main"]/iframe')
+  const srcHandle = await iframeNode.getProperty('src')
+  const src = await srcHandle.jsonValue()
+
+  return await waitForFrame(src)
+}
+
+const findInGrid = async (page, text) => {
+  return await page.waitForXPath(`//*[@role="grid"][contains(normalize-space(.),"${text}")]`)
+}
+
+const exactlyFindClickable = (page, text) => {
+  return page.waitForXPath(`(//a | //*[@role="button"] | //button)[text()="${text}"]`)
```

Comment:
We might want to use something like `normalize-space(.)="${text}"` here to correctly handle nested text.